dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -4581,16 +4581,16 @@ public void validateContentlet(Contentlet contentlet,List<Category> cats)throws
         if(contentlet == null){
             throw new DotContentletValidationException("The contentlet must not be null.");
         }
-        String stInode = contentlet.getStructureInode();
-        if(!InodeUtils.isSet(stInode)){
+        String contentTypeId = contentlet.getContentTypeId();
+        if(!InodeUtils.isSet(contentTypeId)){
             throw new DotContentletValidationException("Contentlet ["+ (contentlet != null ? contentlet.getIdentifier() : "Unknown/New")
                     + "] is not associated to any Content Type.");
         }
-        Structure st = CacheLocator.getContentTypeCache().getStructureByInode(contentlet.getStructureInode());
-        if(Structure.STRUCTURE_TYPE_FILEASSET == st.getStructureType()){
+        Structure contentType = CacheLocator.getContentTypeCache().getStructureByInode(contentlet.getContentTypeId());
```

Comment:
@jcastro-dotcms you can use here this:
`final ContentType contentType = APILocator.getContentTypeAPI().find(contentlet.getStructureInode())`
and in the IFs use:
`BaseContentType.FILEASSET == contentType.baseType()`